resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/v1/KubernetesRestProtocolMessages.scala

```diff
@@ -35,6 +37,9 @@ case class KubernetesCreateSubmissionRequest(
     driverPodKubernetesCredentials: KubernetesCredentials,
     uploadedJarsBase64Contents: TarGzippedData,
     uploadedFilesBase64Contents: TarGzippedData) extends SubmitRestProtocolRequest {
+  @JsonIgnore
+  override val messageType: String = s"kubernetes.v1.${Utils.getFormattedClassName(this)}"
```

Comment:
is this to support compat across a span of versions?  Maybe a new rest service with an old launcher?